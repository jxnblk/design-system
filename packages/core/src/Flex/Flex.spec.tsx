import React from 'react'

import { Flex } from '..'

describe('Flex', () => {
  test('renders', () => {
    const flex = rendererCreateWithTheme(<Flex />).toJSON()
    expect(flex).toMatchSnapshot()
  })

  test('alignItems prop', () => {
    const flex = rendererCreateWithTheme(<Flex alignItems='center' />).toJSON()
    expect(flex).toMatchSnapshot()
    expect(flex).toHaveStyleRule('align-items', 'center')
  })

  test('justifyContent prop', () => {
    const flex = rendererCreateWithTheme(<Flex justifyContent='space-between' />).toJSON()
    expect(flex).toMatchSnapshot()
    expect(flex).toHaveStyleRule('justify-content', 'space-between')
  })

  test('flexWrap prop', () => {
    const flex = rendererCreateWithTheme(<Flex flexWrap='wrap' />).toJSON()
    expect(flex).toMatchSnapshot()
    expect(flex).toHaveStyleRule('flex-wrap', 'wrap')
  })

  describe('deprecated prop types', () => {
    let consoleError
    beforeEach(() => {
      consoleError = console.error
      console.error = jest.fn()
    })
    afterEach(() => (console.error = consoleError))

    test('shims the deprecated `align` prop and warns', () => {
      const json = rendererCreateWithTheme(<Flex align='center' />).toJSON()
      expect(json).toHaveStyleRule('align-items', 'center')
      expect(console.error).toHaveBeenCalledWith(
        'Warning: Failed %s type: %s%s',
        'prop',
        'The `align` prop is deprecated and will be removed in a future release. Please use `alignItems` instead.',
        expect.any(String)
      )
    })

    test('shims the deprecated `wrap` prop and warns', () => {
      const json = rendererCreateWithTheme(<Flex wrap />).toJSON()
      expect(json).toHaveStyleRule('flex-wrap', 'wrap')
      expect(console.error).toHaveBeenCalledWith(
        'Warning: Failed %s type: %s%s',
        'prop',
        'The `wrap` prop is deprecated and will be removed in a future release. Please use `flexWrap` instead.',
        expect.any(String)
      )
    })

    test('shims the deprecated `justify` prop and warns', () => {
      const json = rendererCreateWithTheme(<Flex justify='space-between' />).toJSON()
      expect(json).toHaveStyleRule('justify-content', 'space-between')
      expect(console.error).toHaveBeenCalledWith(
        'Warning: Failed %s type: %s%s',
        'prop',
        'The `justify` prop is deprecated and will be removed in a future release. Please use `justifyContent` instead.',
        expect.any(String)
      )
    })
  })
})
