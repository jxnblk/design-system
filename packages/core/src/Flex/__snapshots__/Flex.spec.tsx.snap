// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex alignItems prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Flex flexWrap prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Flex justifyContent prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Flex renders 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
